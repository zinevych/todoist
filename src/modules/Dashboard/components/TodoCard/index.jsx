import { makeStyles } from "@material-ui/core/styles";
import {
  Card,
  CardHeader,
  IconButton,
  CardContent,
  Typography,
  Grid,
} from "@material-ui/core";
import Close from "@material-ui/icons/Close";
import { connect } from "react-redux";

import PropTypes from "prop-types";

import { removeTodoItem } from "../../actions/actionTypes";

const useStyles = makeStyles((theme) => ({
  card: {
    maxWidth: 300,
    // margin: "auto",
    transition: "0.3s",
    boxShadow: "0 8px 40px -12px rgba(0,0,0,0.3)",
    "&:hover": {
      boxShadow: "0 16px 70px -12.125px rgba(0,0,0,0.3)",
    },
  },
  content: {
    textAlign: "left",
    padding: theme.spacing.unit * 3,
  },
}));

const TodoCard = ({ title, description, type, id, removeTodoItemAction }) => {
  const classes = useStyles();

  const handleRemove = () =>
    removeTodoItemAction({
      title,
      desc: description,
      type,
      id,
    });

  return (
    <Grid item>
      <Card className={classes.card}>
        <CardHeader
          action={
            <IconButton aria-label="settings" onClick={handleRemove}>
              <Close />
            </IconButton>
          }
          title={title}
        />
        <CardContent className={classes.content}>
          <Grid item>
            <Typography className="MuiTypography--subheading" variant="caption">
              {description}
            </Typography>
          </Grid>
          <Grid item>
            <Typography className="MuiTypography--subheading" variant="caption">
              {type}
            </Typography>
          </Grid>
        </CardContent>
      </Card>
    </Grid>
  );
};

TodoCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  removeTodoItemAction: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  removeTodoItemAction: (todoItem) =>
    dispatch({ type: removeTodoItem, payload: todoItem }),
});

export default connect(null, mapDispatchToProps)(TodoCard);
