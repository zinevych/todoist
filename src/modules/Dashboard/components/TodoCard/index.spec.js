/* eslint-disable no-undef */
import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import TodoCard from "./index";

const mockStore = configureMockStore();
const store = mockStore({});

describe(`TodoCard rendering`, () => {
  it("renders the TodoCard", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <TodoCard id={1} type="personal" description="test" title="test" />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
