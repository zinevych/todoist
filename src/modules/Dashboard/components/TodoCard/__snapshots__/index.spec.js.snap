// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoCard rendering renders the TodoCard 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          test
        </span>
      </div>
      <div
        className="MuiCardHeader-action"
      >
        <button
          aria-label="settings"
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiCardContent-root makeStyles-content-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <span
          className="MuiTypography-root MuiTypography--subheading MuiTypography-caption"
        >
          test
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <span
          className="MuiTypography-root MuiTypography--subheading MuiTypography-caption"
        >
          personal
        </span>
      </div>
    </div>
  </div>
</div>
`;
